### PR TITLE
[WIP] enhancements and bugfixes for new logger

### PR DESCRIPTION
**Pull request by kd0aij**
@dagar I just rebased my working branch on newLogger2; 
commits 	0fd22ca and 	1c98ca6 might be useful to you since they fix problems I was encountering with the logger

### CODE REVIEW

**Comment by kd0aij** [comment]
allocating a 32K log buffer in this branch succeeds, and shows 3824 bytes of free memory immediately after boot, but most commands like "top" result in a response like
```
nsh: top: command not found
```
I find that I have to reduce the log buffer size by at least 10K to avoid that problem

**Comment by LorenzMeier** [comment]
@julianoes Note the change in commander which works around the alignment issues you seem to have faced on Snapdragon as well.

Before this all goes onto master I want to be sure the topic generation change is not introducing alignment drama. @dagar Have you time for this today or on the weekend?

**Comment by dagar** [comment]
I'm going to remove any possibility of uorb autogen introducing alignment issues. I just need to get the message serialization generated code working well. It was pretty straightforward until I got to orb messages containing other orb messages (position_setpoint_triplet).

**Comment by kd0aij** [comment]
@dagar @tumbili I've added code to the new logger to write the git version and hardware arch. to the logfile. I'll merge from this branch to newLoggerX after Daniel is finished with the uorb_autogen changes.

**Comment by kd0aij** [comment]
Note that the enhancements here will be merged into https://github.com/PX4/Firmware/pull/4365 once branch uorb_autogen (https://github.com/PX4/Firmware/pull/4366) is finished and merged

File: src/modules/logger/logger.cpp [hunk 1]
```diff
@@ -718,7 +718,7 @@ void Logger::write_info(const char *name, const char *value)
 
 	/* construct format key (type and name) */
 	size_t vlen = strlen(value);
-	msg->key_len = snprintf(msg->key, sizeof(msg->key), "char[%d] %s", vlen, name);
+	msg->key_len = snprintf(msg->key, sizeof(msg->key), "char[%u] %s", (unsigned)vlen, name);
```
Comment (dagar) [inline src/modules/logger/logger.cpp]: Try %zu without the cast.

Comment (kd0aij) [inline src/modules/logger/logger.cpp]: Thanks, are the posix and NuttX builds both C99?

Comment (dagar) [inline src/modules/logger/logger.cpp]: Both posix and nuttx are c std=gnu99 and c++ std=gnu++0x

**Comment by kd0aij** [comment]
FlightPlot annotations showing changes in parameter values:

param_changes

**Comment by dagar** [comment]
Are those annotations automatic?

**Comment by kd0aij** [comment]
Right now "unsaved" parameters are written to the log each time the parameter_update topic is updated. I've modified ULogReader to construct a List of timestamped parameter values for each parameter which changes, using the most recently read timestamp. The display is currently just hacked into FlightPlot.generateSeries, and it displays everything in the List. That's probably good enough, since parameter changes shouldn't be too frequent or numerous.

**Pull request merged**